components/Tabs/Tabs.js

```diff
@@ -1,58 +1,76 @@
+    // If the data is 'all' then select all cards regardless of their 
+    //data attribute
+    if(`${this.tabData}` === 'all'){
+      this.cards = document.querySelectorAll('.card');
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
     }
+   
 
-    // Map over the cards array and convert each card element into a new instance of the TabCard class. 
+    // Map over the cards array and convert each card element into a 
+    //new instance of the TabCard class. 
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(item => new TabCard(item));
+    
+    
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener('click', ()=>{this.selectTab()});
   }
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    const tabs = document.querySelectorAll();
-    // Iterate through the NodeList removing the .active-tab class from each element
-    tabs.forEach()
+    const tabs = document.querySelectorAll('.tab');
+    // Iterate through the NodeList removing the .active-tab class 
+    //from each element
+    tabs.forEach(function(item){
+      item.classList.remove('active-tab');
+    })
     // Add a class of ".active-tab" to this.element
-    this.element;
+    this.element.classList.add('active-tab');
 
 
     // Select all of the elements with the .card class on them
-    const cards = ;
-    // Iterate through the NodeList setting the display style each one to 'none'
-    cards.forEach()
-    // Notice we are looping through the this.cards array and invoking selectCard() from the TabCard class, nothing to update here
+    const cards = document.querySelectorAll('.card');
+    // Iterate through the NodeList setting the display style on each  
+    //one to 'none'
+    cards.forEach(function(item){
+      item.style = 'display: none';
+    })
+    // Notice we are looping through the this.cards array and invoking 
+    //selectCard() from the TabCard class, nothing to update here
     this.cards.forEach(card => card.selectCard());
   }
 }
 
 class TabCard {
   constructor(element){
     // Assign this.element to the passed in element.
-    this.element;
+    this.element = element;
   }
   selectCard(){
     // Update the style of this.element to display = null
-    this.element;
+    this.element.style = 'display: null';
   }
 
 }
 
 // Create a reference to all ".tab" classes
-let tabs = document.querySelectorAll();
-// Map over the array and convert each tab reference into a new TabLink object.  Pass in the tab object to the Tabs class.
-tabs = Array.from(tabs).map()
+let tabs = document.querySelectorAll('.tab');
+// Map over the array and convert each tab reference into a new TabLink 
+//object.  Pass in the tab object to the Tabs class.
+tabs = Array.from(tabs).map(item => new TabLink(item))
```

Comment:
Great job with the Js. It looked perfect to me, keep it up and make sure to review this component structure before we get into react.